### DISCUSSION TITLE
How to mark workflow as faulted using Event Hub?

### DISCUSSION THREAD

**Discussion by rairafaqatali** [elsa-workflows/elsa-core#1637] [Q&A]
Hi @sfmskywalker 
**Scenario:**
I have a parent-child workflow. The parent triggers the child's workflow and waits for its completion. In child workflow, I have a custom activity that runs a data brick job. When a child runs the data brick job, it returns the run id and child workflow completed successfully. 

On the backend, the data brick job continues its working and it may be complete or fail. The data brick job publishes an event that is queued in Event Hub.

Now, My question starts here:

- How can I catch that signal using Elsa? 
- How can I mark child workflow complete or fail, based on the signal?
**- How to mark child workflow as faulted?**

Basically, I want to mark the child workflow as either failed or completed based on signal? Is there any way to capture two signals with Signal Received activity, if the signal is completed, mark the child as Completed and if the signal received is failed, mark the child workflow as faulted?

Thanks

**Comment by sfmskywalker** [comment]
That's a great scenario.

The way I might go about this using Signals would be as follows:

- From the child workflow, use your custom activity that runs the data brick job and connect it to a `Fork` activity which branches into two `SignalReceived` activities*: one waiting on a "Completed" signal and the other on "Failed". The child workflow is now suspended on both these activities.
- Once the brick job is done, it publishes an event to a queue on Event Hub as you mentioned. Which means that you need an Event Hub queue listener in your application. From this listener, trigger a signal using `ISignaler.TriggerSignalAsync`. The signal to trigger depends on the status of the brick job: completed or failed.
- If the Failed signal is triggered, you want to fault the workflow. There's currently no built-in activity that enables you to do so, but it's easy to create a custom activity that does this. All the activity would have to do is return a `Fault` result.

**Reply by fulganta** [reply]
Hi,
In my Fault activity I return Fault with some comment.
`return Fault("Error get access /quote/CorrelationId endpoint");`
So it will make fail my workflow. But I don't know where the error message is displayed.
All I can see is that the activity has been executed and there was  problem.
What's a proper way to display errors?

![image](https://user-images.githubusercontent.com/45927310/137420398-36964d1b-0376-4611-90ee-70fcc0e32be7.png)

**Reply by rairafaqatali** [reply]
Hi, @sfmskywalker 
I have exactly followed the same steps you mentioned. I have written down an activity that returns the Fault('message'). In the case of Completed, I do nothing and workflow status changed from suspended to Finished. But, in the case of the Failed message, I have added my custom activity that marks the workflow as failed. 

The issue is: Activity gets failed, Workflow Time FaultedAt updated but workflow status still remains Suspended. It's not updating the workflow status from Suspended to Faulted.

Kindly let me know what I'm missing?

**Reply by sfmskywalker** [reply]
I think it might be an issue with the engine, the way it (does not) handle faults. Will look into it.
